build.gradle

```diff
@@ -29,6 +29,9 @@ ext {
 
 repositories { mavenCentral() }
 
+def javaVersion = JavaLanguageVersion.of(System.env.JAVA_VERSION ?: 11)
+java.toolchain.languageVersion = javaVersion
```

Comment:
Should we do this within `java`?

```
java {
  toolchain.languageVersion = JavaLanguageVersion.of(...)

  if (toolchain.languageVersion.canCompileOrRun(15)) { ...
```

From [Gradle docs](https://docs.gradle.org/current/userguide/toolchains.html#sec:consuming).

Reply:
Since this project uses script plugins, I preferred doing this before they were applied to avoid any future ordering issues if they eagerly resolved their configuration to a fixed state. In reality this doesn't matter as your scripts do not appear to rely on any of this state in a problematic way. You are certainly welcome to move it for stylistic formatting purposes.

Long term, I'd recommend using precompiled script plugins to avoid quirky surprises that can arise from embedding scriptlets compared to true plugins (e.g. [java-library.caffeine.gradle.kts](https://github.com/ben-manes/caffeine/blob/master/gradle/plugins/src/main/kotlin/lifecycle/java-library.caffeine.gradle.kts)).